dss/util/es.py

```diff
@@ -34,28 +39,68 @@ def __call__(self, r):
         return r
 
 
+class ElasticsearchServer:
+    def __init__(self, startup_timeout_seconds: int=30) -> None:
+        elasticsearch_binary = os.getenv("DSS_TEST_ES_PATH")
+        self.tempdir = tempfile.TemporaryDirectory()
+        while True:
+            port = random.randint(1024, 65535)
```

Comment:
If you're using random port IDs It would be a good idea, once you have picked a new port number, to connect to it and ensure you get and ECONNREFUSED before attempting to use it.